bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/It600Handler.java

```diff
@@ -0,0 +1,265 @@
+                return;
+            }
+            // device is missing properties
+            var deviceProperties = findDeviceProperties().stream().map(DeviceProperty::getName).toList();
+            var result = new ArrayList<>(REQUIRED_CHANNELS);
+            result.removeAll(deviceProperties);
+            if (!result.isEmpty()) {
+                updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                        "Device with DSN " + dsn + " is missing required properties: " + String.join(", ", result));
+                return;
+            }
+        } catch (Exception e) {
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, "Error when connecting to Salus Cloud!");
+            return;
+        }
+
+        // done
+        updateStatus(ONLINE);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        var id = channelUID.getId();
+        switch (id) {
+            case TEMPERATURE:
+                handleCommandForTemperature(channelUID, command);
+                break;
+            case EXPECTED_TEMPERATURE:
+                handleCommandForExpectedTemperature(channelUID, command);
+                break;
+            case WORK_TYPE:
+                handleCommandForWorkType(channelUID, command);
+                break;
+            default:
+                logger.warn("Unknown channel `{}` for command `{}`", id, command);
+        }
+    }
+
+    private void handleCommandForTemperature(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            // only refresh commands are supported for temp channel
+            return;
+        }
+
+        findLongProperty("ep_9:sIT600TH:LocalTemperature_x100", "LocalTemperature_x100")
+                .map(DeviceProperty.LongDeviceProperty::getValue).map(BigDecimal::new)
+                .map(value -> value.divide(ONE_HUNDRED, new MathContext(5, HALF_EVEN))).map(DecimalType::new)
+                .ifPresent(state -> updateState(channelUID, state));
+    }
+
+    private void handleCommandForExpectedTemperature(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            findLongProperty("ep_9:sIT600TH:HeatingSetpoint_x100", "HeatingSetpoint_x100")
+                    .map(DeviceProperty.LongDeviceProperty::getValue).map(BigDecimal::new)
+                    .map(value -> value.divide(ONE_HUNDRED, new MathContext(5, HALF_EVEN))).map(DecimalType::new)
+                    .ifPresent(state -> updateState(channelUID, state));
+            return;
+        }
+
+        if (command instanceof DecimalType || command instanceof QuantityType) {
+            BigDecimal rawValue;
+            if (command instanceof DecimalType typedCommand) {
+                rawValue = typedCommand.toBigDecimal();
+            } else {
+                var typedCommand = (QuantityType<?>) command;
+                rawValue = typedCommand.toBigDecimal();
+            }
```

Comment:
```suggestion
        BigDecimal rawValue = null;
        if (command instanceof QuantityType commandAsQuantityType) {
            rawValue = commandAsQuantityType.toBigDecimal();
        } else if (command instanceof DecimalType commandAsDecimalType) {
            rawValue = commandAsDecimalType.toBigDecimal();
        }
        
        if (rawValue != null) {
```

Beesides this minor improvement, i aslo wonder if you don't need to check the quantity unit/scale and convert it to whatever the remote service expects?

Reply:
> if you don't need to check the quantity unit/scale and convert it to whatever the remote service expects?

can you explain it?

BTW your code has `Raw use of parameterized class 'QuantityType'` problem

Reply:
Updated the suggestion, with the diamond operator. It is a java17 feauture that when checking the instanceof, it also assinggs the variable. This save an additional assignment.

Reply:
Ah, sorry i explained thee wrong part :-) 

If i'm correct, the current implementation sends different `value` to the remote service depending on the user sending a quantitytype command in Fahrenheit or Celsius?

If the remote service expects all temperature commands to be in Celsius i would expect:

```
commandAsQuantityType.toUnit(SIUnits.CELSIUS).toBigDecimal()
```